TestNoLinkTesting breaks release build

https://circleci.com/gh/cockroachdb/cockroach/2179

The issue is that for the Docker deploy image, we build all the tests and then put them into a container that does not have the sources or Go.

TestNoLinkTesting calls out to the runtime to calculate the build dependencies, which isn't possible in that context. We should probably move that test to its own file and not build it with the `release` tag.

```
executing ../build/cli.test
--- FAIL: TestNoLinkTesting (0.00s)
	flags_test.go:47: cannot find package "github.com/cockroachdb/cockroach" in any of:
			/usr/src/go/src/github.com/cockroachdb/cockroach (from $GOROOT)
			($GOPATH not set)
```

--- a/server/cli/flags_test.go
+++ b/server/cli/flags_test.go
@@ -38,6 +38,10 @@ func TestStdFlagToPflag(t *testing.T) {
 }
 
 func TestNoLinkTesting(t *testing.T) {
+	if build.Default.GOPATH == "" {
+		t.Skip("GOPATH isn't set")
+	}
+
 	imports := make(map[string]struct{})
 
 	var addImports func(string)

